test/interoperability/interop_test_suite.py

```diff
@@ -0,0 +1,537 @@
+        primary_dss: str,
+        all_other_dss: List[str],
+    ) -> None:
+        """Can query all Subscriptions in area from all DSSs."""
+        all_dss = [primary_dss] + all_other_dss
+        all_sub_1 = set()
+        for index in range(len(all_dss)):
+            all_sub_1.add(self.context[f"sub_1_{index}_uuid"].key)
+        for index, dss in enumerate(all_dss):
+            resp = dss_map[dss].get(f"/subscriptions?area={GEO_POLYGON_STRING}")
+            assert resp.status_code == 200, f"{dss} failed to get SUB_1 by area"
+
+            returned_subs = set([x["id"] for x in resp.json()["subscriptions"]])
+
+            missing_subs = all_sub_1 - returned_subs
+            assert (
+                missing_subs == set()
+            ), f"{dss} returned too few Subscriptions, missing: {missing_subs}"
+
+            extra_subs = returned_subs - all_sub_1
+            assert (
+                extra_subs == set()
+            ), f"{dss} returned too many Subscriptsions, extra: {extra_subs}"
+
+    def testStep5(
+        self, dss_map: Dict[str, clients.DSSClient], primary_dss: str, **kwargs
+    ) -> None:
+        """Can modify ISA in primary DSS, ISA modification triggers 
+        subscription notification requests"""
+        isa_1_uuid = self.context["isa_1_uuid"].key
+        resp = dss_map[primary_dss].get(f"/identification_service_areas/{isa_1_uuid}")
+        assert (
+            resp.status_code == 200
+        ), f"Failed to find ISA_1 in Primary DSS: {primary_dss}"
+        data = resp.json()["service_area"]
+
+        time_end = datetime.datetime.utcnow() + datetime.timedelta(
+            seconds=SHORT_WAIT_SEC
+        )
+        put_resp = dss_map[primary_dss].put(
+            f"/identification_service_areas/{isa_1_uuid}/{data['version']}",
+            json={
+                "extents": {
+                    "spatial_volume": {
+                        "footprint": {"vertices": VERTICES},
+                        "altitude_lo": 20,
+                        "altitude_hi": 400,
+                    },
+                    "time_start": data["time_start"],
+                    "time_end": time_end.strftime(DATE_FORMAT),
+                },
+                "flights_url": "https://example.com/dss",
+            },
+        )
+        assert put_resp.status_code == 200, f"Failed to update ISA_1 with new end time"
+
+        updated_data = put_resp.json()
+        assert updated_data["service_area"]["time_end"] == time_end.strftime(
+            DATE_FORMAT
+        ), f"Unsuccessful Update; no change to end time"
+
+    def testStep6(
+        self, dss_map: Dict[str, clients.DSSClient], primary_dss: str, **kwargs
+    ) -> None:
+        """Can delete all Subscription in primary DSS"""
+        dss = dss_map[primary_dss]
+        for name, context in self.context.items():
+            if name.startswith("sub_1_"):
+                resp = dss.delete(f"/subscriptions/{context.key}/")
```

Comment:
This should fail; no `version` parameter is supplied in the path

Reply:
Created #153